File: 1903-largest-odd-number-in-string/1903-largest-odd-number-in-string.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    string largestOddNumber(string num) {
        string largeNumSTR = "";
        string largeNum = "";
        for (char x: num){
            largeNumSTR += x;
            if((x - 48)%2 == 1 ){
                largeNum = largeNumSTR;
            }
        }        
        return largeNum;
    }
};

File: 1903-largest-odd-number-in-string/1903-largest-odd-number-in-string.py
class Solution:
    def largestOddNumber(self, num: str) -> str:
        idx = 0 
        for i in range(len(num)):
            if int(num[i]) % 2 == 1:
                idx = i
        if idx == 0 and int(num[idx]) % 2 == 1:
            return num[idx]
        elif idx == 0:
            return ""
        else:
            return num[:idx+1]
        
        
